File: 2358-number-of-ways-to-split-array/number-of-ways-to-split-array.cpp
class Solution {
public:
    int waysToSplitArray(vector<int>& nums) {
        long long  totSum = 0;
        for(int i:nums) totSum += i;
        long long  currSum = 0;
        int cnt = 0;
        for(int i=0;i<nums.size()-1;i++){
            currSum += nums[i];
            if(currSum >= (totSum-currSum)) cnt++;
        }
        return cnt;
    }
};

File: 2358-number-of-ways-to-split-array/number-of-ways-to-split-array.rs
impl Solution {
    pub fn ways_to_split_array(nums: Vec<i32>) -> i32 {
        let mut totSum:i64 = 0;
        let mut currSum : i64 = 0;
        for i in 0..nums.len(){
            totSum += nums[i] as i64;
        }
        let mut cnt:i32 = 0;
        for i in 0..nums.len()-1{
            currSum += nums[i] as i64;
            if currSum >= totSum-currSum{
                cnt += 1;
            }
        }
        cnt
    }
}